### PR TITLE
Add `inv compile` task, reuse it for CI jobs

### LINKED ISSUE #13: Add helper "compile" Invoke task and change CI build scripts to use it

### PR DESCRIPTION
**Pull request by tadeu**
Resolves #13

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -1,71 +1,153 @@
...
+        cmake_generator="Ninja",
+        config=config,
+    )
+    build_command = strip_and_join(f"""
+        cmake
+            --build .
+            --target install
+            --config {config}
+            --
+                {f"-j {number_of_jobs}" if number_of_jobs >= 0 else ""}
```
Comment (nicoddemus) [inline tasks.py]: Hehehe

File: tasks.py [hunk 2]
```diff
@@ -1,71 +1,153 @@
...
-            -DBOOST_INCLUDE_DIR="{CMAKE_INCLUDE_PATH}"
-            -DCMAKE_INSTALL_PREFIX="{relative_artifacts_dir.as_posix()}"
-            -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON
-            "-DTHIRDPARTY_COMMON_ARGS=-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON"
-            "{str(relative_root_dir)}"
-        """))
+    commands = [cmake_command, build_command]
+
+    if sys.platform.startswith('win'):
+        commands.insert(0, r'"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" amd64')
```
Comment (nicoddemus) [inline tasks.py]: I suggest to ensure that this path exists, and fail hard if it doesn't

File: tasks.py [hunk 3]
```diff
@@ -1,71 +1,153 @@
...
+
+    os.chdir(build_dir)
+    c.run(" && ".join(commands))
```
Comment (nicoddemus) [inline tasks.py]: Hmm shame `c.run` doesn't receive a `cwd` argument :(

**Comment by nicoddemus** [comment]
Great work!

File: tasks.py (hunk 2 above)
Comment (tadeu) [reply tasks.py]: - [ ] Ensure PATH exists

**Pull request merged**